GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -634,35 +639,29 @@ - (void)authorizationRequestWithOptions:(GIDSignInInternalOptions *)options comp
   if (@available(iOS 14.0, *)) {
     if (_appCheck) {
       shouldCallCompletion = NO;
-      GIDActivityIndicatorViewController *activityVC =
-          [[GIDActivityIndicatorViewController alloc] init];
-      [options.presentingViewController presentViewController:activityVC
-                                                     animated:true
-                                                   completion:^{
-        // Ensure that the activity indicator shows for at least 1/2 second to prevent "flashing"
-        // TODO: Re-implement per: https://github.com/google/GoogleSignIn-iOS/issues/329
-        dispatch_time_t halfSecond = dispatch_time(DISPATCH_TIME_NOW, NSEC_PER_MSEC / 2);
-        dispatch_after(halfSecond, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-          [self->_appCheck getLimitedUseTokenWithCompletion:
-              ^(id<GACAppCheckTokenProtocol> _Nullable token, NSError * _Nullable error) {
-            if (token) {
-              additionalParameters[kClientAssertionTypeParameter] =
-                  kClientAssertionTypeParameterValue;
-              additionalParameters[kClientAssertionParameter] = token.token;
-              OIDAuthorizationRequest *request =
-                  [self authorizationRequestWithOptions:options
-                                   additionalParameters:additionalParameters];
-              [activityVC.activityIndicator stopAnimating];
-              [activityVC dismissViewControllerAnimated:YES completion:nil];
-              completion(request, nil);
-              return;
-            }
-            [activityVC.activityIndicator stopAnimating];
-            [activityVC dismissViewControllerAnimated:YES completion:nil];
-            completion(nil, error);
-            return;
+      UIViewController *presentingVC = options.presentingViewController;
+      if (!_timedLoader) {
+        _timedLoader = [[GIDTimedLoader alloc] initWithPresentingViewController:presentingVC];
+      }
+      if (_timedLoader.animationStatus != GIDTimedLoaderAnimationStatusAnimating) {
```

Comment:
Do we need this check if GIDTimedLoader does that check anyway in the implementation of startTiming?

Reply:
Nope!